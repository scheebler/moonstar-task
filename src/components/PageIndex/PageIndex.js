import React from 'react';
import PageIndexContainer from '../PostsList/styled/PageIndexContainer';

const PageIndex = ({index, currentPage, setCurrentPage }) => {
	return (
		<PageIndexContainer onClick={() => setCurrentPage(index)} active={index === currentPage}>
			{index}
		</PageIndexContainer>
	);
};

export default PageIndex;