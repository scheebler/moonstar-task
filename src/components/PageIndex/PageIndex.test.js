import React from 'react';
import PageIndex from './PageIndex';
import renderer from 'react-test-renderer';

describe('Jest PageIndex component', () => {
	it('Test active PageIndex', () => {
		const tree = renderer.create(<PageIndex index={1} currentPage={1} setCurrentPage={jest.fn()} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Test inactive PageIndex', () => {
		const tree = renderer.create(<PageIndex index={1} currentPage={2} setCurrentPage={jest.fn()} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});