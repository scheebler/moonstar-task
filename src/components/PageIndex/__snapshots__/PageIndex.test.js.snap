// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest PageIndex component Test active PageIndex 1`] = `
<div
  className="sc-beqWaB kcmzco"
  onClick={[Function]}
>
  1
</div>
`;

exports[`Jest PageIndex component Test inactive PageIndex 1`] = `
<div
  className="sc-beqWaB fhCxnn"
  onClick={[Function]}
>
  1
</div>
`;
